### ISSUE TITLE
onic ion-header-bar button-right width not calculated correctly when replacing button content using directive

### ISSUE THREAD

**Issue by leschirmeur** [driftyco/ionic#4912]
When using an Ionic ion-header-bar and replacing button-right text using angular-translate as follows, the button-right width is not calculated properly:

```
<ion-header-bar class="bar-light">
        <a class="button button-icon icon slicon slicon-close" ui-sref="start"></a>
        <h1 class="title" translate>register.finishprofile.navbar.label.welcome</h1>
        <button class="button button-clear button-positive button-strong" type="submit" translate>common.button.next</button>
```

However, if passing the angular-translate translation id as an attribute value, the button width is calculated correctly:
````
<ion-header-bar class="bar-light">
        <a class="button button-icon icon slicon slicon-close" ui-sref="start"></a>
        <h1 class="title" translate>register.finishprofile.navbar.label.welcome</h1>
        <button class="button button-clear button-positive button-strong" type="submit" translate="common.button.next"></button>
    </ion-header-bar>
```

So, to me it seems that Ionic takes the length of the translation id if the id is not set as an attribute to determine button width, BEFORE the translation id is replaced by angular-translate.

In other words, the content of the button element gets replaced by the translate directive. The button width however is calculated before this replacement takes place, leading to incorrect button dimensions.
